### PR TITLE
Feat/termination fees

### LINKED ISSUE #437: Implement fault and termination penalty calculations
The fault fee and sector penalty depend on the instantaneous block reward value (from reward actors) and the total committed power (from power actor) during the epoch the fault occurs. Each is a simple multiple of these.

- `BR(t) = CurrEpochReward(t) * SectorQualityAdjustedPower * EpochsInDay  / TotalNetworkQualityAdjustedPower(t)`
- `FF(t) = a*BR(t)`, with `a` = 2.14.
- `SP(t) = 5BR(t)`

The sector termination fee additionally depends on the initial pledge calculated _when the sector was committed_. The intent is to calculate the block reward that was earned by the sector during its life (up to 180 days max), if the block reward had been fixed at the value when the sector was committed. This is then bounded below by `SP` calculated at the termination epoch.

- `TF = max(SP(t), InitialPledge + BR(StartEpoch)*min(SectorAgeInDays, 180))`

Note that this requires recording the block reward value when the sector is committed. We can do this in two ways:
- record the raw BR
- record the initial pledge, from which BR might be calculable if it's a strict function of BR

For now, I propose recording *both*. It will be easy to remove one if it turns out to be redundant.

Consider doing #419 first to ease implementation.

### PR DESCRIPTION
**Pull request by acruikshank**
closes #437

### Motivation

While the infrastructure for charging sector termination fees exists, the actual calculation has not been implemented. This PR implements that calculation and tests.

### Proposed Changes

1. Implement the termination fee formula based on the initial pledge of the sector.
2. Augment test to show correct fee has been sent to burnt funds and pledge has been updated in power actor
3. Clean up miner actor harness by adding `networkQAPower` and `networkRawPower` as instance variables with defaults.

### CODE REVIEW

File: actors/builtin/miner/monies.go [hunk 1]
```diff
@@ -44,9 +44,16 @@ func PledgePenaltyForUndeclaredFault(epochTargetReward abi.TokenAmount, networkQ
 }
 
 // Penalty to locked pledge collateral for the termination of a sector before scheduled expiry.
-func pledgePenaltyForTermination(s *SectorOnChainInfo, epochTargetReward abi.TokenAmount, networkQAPower abi.StoragePower, qaSectorPower abi.StoragePower) abi.TokenAmount {
-	//TODO #437 compute correct termination fee using initial pledge value from SectorOnChainInfo
-	return big.Zero() // PARAM_FINISH
+func PledgePenaltyForTermination(s *SectorOnChainInfo, currEpoch abi.ChainEpoch, epochTargetReward abi.TokenAmount, networkQAPower abi.StoragePower, qaSectorPower abi.StoragePower) abi.TokenAmount {
+	// max(SP(t), InitialPledge + BR(StartEpoch)*min(SectorAgeInDays, 180)) where BF(StartEpoch)=InitialPledge/InitialPledgeFactor
```
Comment (ZenGround0) [inline actors/builtin/miner/monies.go:48]: nit: BF should be BR I think?

File: actors/builtin/miner/monies.go [hunk 2]
```diff
@@ -44,9 +44,16 @@ func PledgePenaltyForUndeclaredFault(epochTargetReward abi.TokenAmount, networkQ
...
+	sectorAge := big.NewInt(int64(minEpoch(currEpoch-s.Activation, 180*builtin.EpochsInDay)))
+	return big.Max(
+		PledgePenaltyForUndeclaredFault(epochTargetReward, networkQAPower, qaSectorPower),
+		big.Add(
+			InitialPledgeFactor,
```
Comment (ZenGround0) [inline actors/builtin/miner/monies.go:53]: From the comment above I would have thought this would be `s.InitialPledge`, not `InitialPledgeFactor`

Comment (anorth) [reply actors/builtin/miner/monies.go:53]: I have been meaning to add unit tests specifically for all these calculation functions. Let's start with this one.

Could you add some basic tests outlining the calculation results? Especially useful might be border cases like 1 epoch, 180days, >180days, and some right-order-of-magnitude numbers for pledge, power etc.

**Comment by acruikshank** [comment]
Good catch @ZenGround0. I've refactored the function a bit to make testing easier and test the major cases. In the process I realized I think I'd made another more subtle mistake. BR represents block reward per day, and I was multiplying it by the sector age in epochs which should be too much by a factor of ~3000. The new version divides by `EpochsInDay` at the end to maintain precision. Please take another looks @anorth and @ZenGround0 if you get a chance.

**Pull request merged**